Model/Sync/Reset/Catalog.php

```diff
@@ -0,0 +1,143 @@
+     * @param SyncDataMain $syncDataMain
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList,
+        SyncDataMain $syncDataMain
+    ) {
+        parent::__construct(
+            $resourceConnection,
+            $config,
+            $cacheTypeList
+        );
+        $this->syncDataMain = $syncDataMain;
+    }
+
+    /**
+     * @return string
+     */
+    public function getYotpoEntityName()
+    {
+        return self::YOTPO_ENTITY_NAME;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        parent::resetSync($storeId);
+        $this->resetCatalogSyncAttributes($storeId);
+        $this->setResetInProgressConfig($storeId, '0');
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getTableResourceNames()
+    {
+        return [self::PRODUCT_SYNC_TABLE, self::CATEGORY_SYNC_TABLE];
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getCronJobCodes()
+    {
+        return self::CRONJOB_CODES;
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    private function resetCatalogSyncAttributes($storeId)
+    {
+        $dataSet = [
+            [
+                'table_name' => 'catalog_category_entity_int',
+                'attribute_code' => CoreConfig::CATEGORY_SYNC_ATTR_CODE
+            ],
+            [
+                'table_name' => 'catalog_product_entity_int',
+                'attribute_code' => CoreConfig::CATALOG_SYNC_ATTR_CODE
+            ],
+        ];
+        foreach ($dataSet as $data) {
+            $connection =   $this->resourceConnection->getConnection();
+            $attributeId = $this->syncDataMain->getAttributeId($data['attribute_code']);
+            $totalCount = $this->getCountOfEntities($data['table_name'], $attributeId, $storeId);
+            $tableName = $this->resourceConnection->getTableName($data['table_name']);
+            while ($totalCount > 0) {
+                if ($totalCount > self::UPDATE_LIMIT) {
+                    $limit = self::UPDATE_LIMIT;
+                    $totalCount -= self::UPDATE_LIMIT;
+                } else {
+                    $limit = $totalCount;
+                    $totalCount = 0;
+                }
+                $updateQuery = sprintf(
+                    'UPDATE `%s` SET `value` = %d WHERE `attribute_id` = %d AND `store_id` = %d AND `value` = 1
+                        ORDER BY `value_id` ASC LIMIT %d',
+                    $this->resourceConnection->getTableName($tableName),
+                    0,
+                    $this->syncDataMain->getAttributeId($data['attribute_code']),
+                    $storeId,
+                    $limit
+                );
```

Comment:
1. Can you explain why did you choose to use an explicit query over here?
2. Why do you `ORDER BY value_id ASC`?

Reply:
Without explicit Query , there is not built in methods in magento to update a table with LIMIT. Otherwise we need to select some LIMITED records first and then update them, which will be performance concerning approach. 

Please let me know if we need to make any changes here.

Reply:
Understood.
What about the second question in the initial comment?

Reply:
@LiorGingi ,

There was no specific reason. It was under the thought that it will update the records sequentially in the order of value_id in the attribute table . We will remove it